x-pack/elastic-agent/pkg/agent/application/upgrade/step_download.go

```diff
@@ -84,18 +84,28 @@ func newDownloader(version string, log *logger.Logger, settings *artifact.Config
 		return downloader.NewDownloader(log, settings)
 	}
 
+	// We can use up to 3 downloaders
```

Comment:
Do we have any unit test covering this function? Looks like this is the bulk of the change in this PR.

Reply:
this part of the agent 7.17 is not exactly testable (the fact that instantiating a downloader fires an http request is a big symptom of that) so I am not sure how much we want to change this just to fix this problem...

Reply:
Ok, your manual test will have to do then.